src/driver/driver.cpp

```diff
@@ -65,6 +66,19 @@ void Driver::PostExecute(DriverStatus status) {
 }
 
 DriverStatus EvolutionDriver::Execute() {
+  // JMM: I want these before output_params_and_exit so we capture as much as possible
+  OutputSignal signal = pinput->GetBoolean("parthenon/job", "run_only_analysis")
+                            ? OutputSignal::analysis
+                            : OutputSignal::none;
+  int perf_cycle_offset = pinput->GetOrAddInteger(
+      "parthenon/time", "perf_cycle_offset", 0,
+      "don't measure performance for some number of initial cycles");
+
+  if (Globals::output_params_and_exit && Globals::my_rank == 0) {
+    pinput->OutputParameterTable(std::cout, std::regex(Globals::params_block_regex));
+    return DriverStatus::complete;
+  }
```

Comment:
The ordering here is a little tricky. This has to be done after setup (so that pgens are captured) but before any actual looping. I chose this conditional logic here.